MetadataExtractor/IO/IndexedReader.cs

```diff
@@ -307,33 +255,77 @@ public ulong GetUInt64(int index)
         /// <exception cref="IOException">the buffer does not contain enough bytes to service the request, or index is negative</exception>
         public float GetS15Fixed16(int index)
         {
-            ValidateIndex(index, 4);
+            Span<byte> bytes = stackalloc byte[4];
+
+            GetBytes(index, bytes);
+
             if (IsMotorolaByteOrder)
             {
-                float res = GetByteInternal(index) << 8 | GetByteInternal(index + 1);
-                var d = GetByteInternal(index + 2) << 8 | GetByteInternal(index + 3);
+                float res = bytes[0] << 8 | bytes[1];
+                var d = bytes[2] << 8 | bytes[3];
                 return (float)(res + d / 65536.0);
             }
             else
             {
                 // this particular branch is untested
-                var d = GetByteInternal(index + 1) << 8 | GetByteInternal(index);
-                float res = GetByteInternal(index + 3) << 8 | GetByteInternal(index + 2);
+                var d = bytes[1] << 8 | bytes[0];
+                float res = bytes[3] << 8 | bytes[2];
                 return (float)(res + d / 65536.0);
             }
         }
 
         /// <exception cref="IOException"/>
-        public float GetFloat32(int index) => BitConverter.ToSingle(BitConverter.GetBytes(GetInt32(index)), 0);
+        public float GetFloat32(int index)
+        {
+#if NET45 || NETSTANDARD1_3
+            return BitConverter.ToSingle(BitConverter.GetBytes(GetInt32(index)), 0);
+#else
+            Span<byte> bytes = stackalloc byte[4];
+
+            GetBytes(index, bytes);
+
+            if (IsMotorolaByteOrder)
+            {
+                bytes.Reverse();
+            }
+
+            return BitConverter.ToSingle(bytes);
+#endif
+        }
 
         /// <exception cref="IOException"/>
-        public double GetDouble64(int index) => BitConverter.Int64BitsToDouble(GetInt64(index));
+        public double GetDouble64(int index)
+        {
+#if NET45 || NETSTANDARD1_3
+            return BitConverter.Int64BitsToDouble(GetInt64(index));
+#else
+            Span<byte> bytes = stackalloc byte[8];
+
+            GetBytes(index, bytes);
+
+            if (IsMotorolaByteOrder)
+            {
+                bytes.Reverse();
+            }
+
+            return BitConverter.ToDouble(bytes);
+#endif
+        }
 
         /// <exception cref="IOException"/>
         public string GetString(int index, int bytesRequested, Encoding encoding)
         {
+#if NET45 || NETSTANDARD1_3
             var bytes = GetBytes(index, bytesRequested);
+
             return encoding.GetString(bytes, 0, bytes.Length);
+#else
+            Span<byte> bytes = bytesRequested < 256 ? stackalloc byte[bytesRequested] : new byte[bytesRequested];
+
+            GetBytes(index, bytes);
+
+            return encoding.GetString(bytes);
```

Comment:
If we bump netframework to 462, we may consider adding a polyfill for Encoding.GetString to have two paths here. This would remain allocation free, and simplify ^.

For example:

```csharp
#if NETFRAMEWORK || NETSTANDARD1_3
internal static class EncodingExtensions
{
    public unsafe static string GetString(this Encoding encoding, ReadOnlySpan<byte> bytes)
    {
        fixed (byte* pBytes = bytes)
        {
            return encoding.GetString(pBytes, bytes.Length);
        }
    }
}
#endif
```

Reply:
That's a great approach and we could use it in a few places.

I'll have a think about bumping the target further. Are there other APIs in 462 you know of that's be good to use?

Reply:
Array.Empty<T> is the only other one I've missed so far. That would allow us to kill Util.Empty. It's possible that the lowering of [] may also be improved on net462.

Reply:
I'm good with bumping to `net462`. 4.5.2 went out of support in 2022. 4.6.2 is supported until 2027. I'll file a different PR for that then rebase and update this one.

Reply:
https://github.com/drewnoakes/metadata-extractor-dotnet/pull/386